lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -262,6 +262,46 @@ ::mlir::LogicalResult mlir::tt::ttir::ReshapeOp::verify() {
   return success();
 }
 
+::mlir::LogicalResult mlir::tt::ttir::UnsqueezeOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType outputType = getOutput().getType();
+  int32_t dim = getDim();
+
+  // Check that the dim is within the bounds of the input tensor
+  if (dim >= inputType.getRank() || dim < 0) {
+    return emitOpError(
+        "Dimension attribute must be within the bounds of the input tensor");
+  }
+
+  // Check that the output tensor has one more dimension than the input tensor
+  if (outputType.getRank() != inputType.getRank() + 1) {
+    return emitOpError(
+        "Output tensor must have one more dimension than the input tensor");
+  }
+
+  // and that the dimension added is of size 1
+  if (outputType.getDimSize(dim) != 1) {
+    return emitOpError("Dimension added must be of size 1");
+  }
+
+  // All dimensions of the input tensor must be the same as the output tensor
+  // except for the dimension added
+  for (int64_t i = 0, j = 0; i < outputType.getRank(); ++i) {
+    if (i == dim) {
+      continue;
+    }
+
+    if (inputType.getDimSize(j) != outputType.getDimSize(i)) {
+      return emitOpError("All dimensions of the input tensor must be the same "
+                         "as the output tensor except for the dimension added");
+    }
+
+    j++;
+  }
```

Comment:
To me personally, this seems cleaner, but it's not a strong opinion so I'll let you make the call :)
```
  bool add_one = 0;
  for (int64_t i = 0; i < inputType.getRank(); ++i) {
    if (i == dim) {
      add_one = true;
    }

    if (inputType.getDimSize(i) != outputType.getDimSize(i + add_one)) {
      return emitOpError("All dimensions of the input tensor must be the same "
                         "as the output tensor except for the dimension unsqueezed");
    }
  }
```

Reply:
@svuckovicTT Not sure if I like that one. What about this one:

```
int64_t j = 0;
for (int64_t i = 0; i < outputType.getRank(); ++i) {
  if (i != dim) {
    if (inputType.getDimSize(j++) != outputType.getDimSize(i)) {
      return emitOpError("All dimensions of the input tensor must be the same "
                         "as the output tensor except for the dimension added");
    }
  }
}
```

Reply:
PS personally I don't like this `j++` which is kinda hidden...

Reply:
Out of that one and the original one, I like the original one better - less nesting is easier on the eyes.